src/main/java/org/opensearch/performanceanalyzer/collectors/telemetry/RTFDisksCollector.java

```diff
@@ -0,0 +1,113 @@
+import org.opensearch.performanceanalyzer.commons.config.overrides.ConfigOverridesWrapper;
+import org.opensearch.performanceanalyzer.commons.metrics.MetricsConfiguration;
+import org.opensearch.performanceanalyzer.commons.metrics.RTFMetrics;
+import org.opensearch.performanceanalyzer.commons.metrics_generator.DiskMetricsGenerator;
+import org.opensearch.performanceanalyzer.commons.metrics_generator.OSMetricsGenerator;
+import org.opensearch.performanceanalyzer.commons.stats.metrics.StatExceptionCode;
+import org.opensearch.performanceanalyzer.commons.stats.metrics.StatMetrics;
+import org.opensearch.performanceanalyzer.config.PerformanceAnalyzerController;
+import org.opensearch.telemetry.metrics.Histogram;
+import org.opensearch.telemetry.metrics.MetricsRegistry;
+import org.opensearch.telemetry.metrics.tags.Tags;
+
+public class RTFDisksCollector extends PerformanceAnalyzerMetricsCollector
+        implements TelemetryCollector {
+
+    private Histogram diskWaitTimeMetrics;
+    private Histogram diskServiceRateMetrics;
+    private Histogram diskUtilizationMetrics;
+    private MetricsRegistry metricsRegistry;
+    private boolean metricsInitialised;
+    private static final Logger LOG = LogManager.getLogger(RTFDisksCollector.class);
+    private PerformanceAnalyzerController performanceAnalyzerController;
+    private ConfigOverridesWrapper configOverridesWrapper;
+
+    public RTFDisksCollector(
+            PerformanceAnalyzerController performanceAnalyzerController,
+            ConfigOverridesWrapper configOverridesWrapper) {
+        super(
+                MetricsConfiguration.CONFIG_MAP.get(RTFDisksCollector.class).samplingInterval,
+                "RTFDisksCollector",
+                StatMetrics.RTF_DISKS_COLLECTOR_EXECUTION_TIME,
+                StatExceptionCode.RTF_DISK_METRICS_COLLECTOR_ERROR);
+        this.metricsInitialised = false;
+        this.performanceAnalyzerController = performanceAnalyzerController;
+        this.configOverridesWrapper = configOverridesWrapper;
+    }
+
+    @Override
+    public void collectMetrics(long startTime) {
+        if (performanceAnalyzerController.isCollectorDisabled(
+                configOverridesWrapper, getCollectorName())) {
+            LOG.info("RTFDisksCollector is disabled. Skipping collection.");
+            return;
+        }
+
+        OSMetricsGenerator generator = OSMetricsGeneratorFactory.getInstance();
+        if (generator == null) {
+            LOG.error("could not get the instance of OSMetricsGeneratorFactory class");
+            return;
+        }
+
+        metricsRegistry = OpenSearchResources.INSTANCE.getMetricsRegistry();
+        if (metricsRegistry == null) {
+            LOG.error("could not get the instance of MetricsRegistry class");
+            return;
+        }
+
+        LOG.debug("Executing collect metrics for RTFDisksCollector");
+
+        initialiseMetricsIfNeeded();
```

Comment:
Do we need to call this again can we not use the constructor for this

Reply:
This cannot be added in constructor because metricsRegistry object is not available by that time. This is already covered in one of the resolved comments.

Reply:
For now should be ok lets see if we can improvise on this as P1